.github/workflows/linux.yml

```diff
@@ -497,3 +497,44 @@ jobs:
         PKG_CONFIG_PATH="$PWD" pip install . -vv --no-build-isolation
         pushd $RUNNER_TEMP
         PYTHON_GIL=0 python -m pytest --pyargs scipy -n2 --durations=10
+  #################################################################################
+  clang_17_build:
+    # Purpose is to check for warnings in builds with latest clang.
+    # We do not run the test suite here.
+    name: Clang-17 build, py3.12
```

Comment:
minor: can you change "build" to "build-only"? That distinguishes the job more. And let's drop `py3.12`, because it's irrelevant.  

I suggest this as the full name: `Clang-17 build-only (-Werror)`